corehq/motech/repeaters/repeater_generators.py

```diff
@@ -739,3 +740,20 @@ def content_type(self):
 
     def get_payload(self, repeat_record, location):
         return json.dumps(location.to_json())
+
+
+class DataSourcePayloadGenerator(BasePayloadGenerator):
+    format_name = 'json'
+    format_label = _('JSON')
+
+    @property
+    def content_type(self):
+        return 'application/json'
+
+    def get_payload(self, repeat_record, docs: list):
```

Comment:
isn't this just a doc and not docs?

Reply:
In `get_payload()` and `payload_doc()` the payload is a list of dictionaries, but in `allowed_to_forward()` the payload is a `DataSourceUpdateLog` instance. This felt surprising, and inconsistent with other Repeater subclasses. So I changed them to all use a `DataSourceUpdateLog` instance. ff853d68d601ee0e8bb65ff365e53b3dd935a5be

Reply:
Nice!